xamarin/realm-tutorial-dotnet/TaskPage.xaml.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using RealmTemplateApp.Models;
+using Realms;
+using Xamarin.Forms;
+using System.ComponentModel;
+
+namespace RealmTemplateApp
+{
+    public partial class TaskPage : ContentPage
+    {
+        private Realm taskRealm;
+        private ObservableCollection<Task> _tasks = new ObservableCollection<Task>();
+
+        public ObservableCollection<Task> MyTasks
+        {
+            get
+            {
+                return _tasks;
+            }
+        }
+
+        public TaskPage()
+        {
+            InitializeComponent();
+        }
+
+        protected override async void OnAppearing()
+        {
+            WaitingLayout.IsVisible = true;
+            try
+            {
+                taskRealm = await Realm.GetInstanceAsync();
+                SetUpTaskList();
+            }
+            catch (Exception ex)
+            {
+                await DisplayAlert("Error Fetching Tasks", ex.Message, "OK");
+            }
+            base.OnAppearing();
+        }
+
+        private void SetUpTaskList()
+        {
+            if (_tasks != null || _tasks.Count == 0)
+            {
+                WaitingLayout.IsVisible = true;
+                _tasks = new ObservableCollection<Task>(taskRealm.All<Task>().ToList());
+                listTasks.ItemsSource = _tasks;
+                WaitingLayout.IsVisible = false;
+            }
+        }
+
+        async void New_Button_Clicked(object sender, EventArgs e)
+        {
+            string result = await DisplayPromptAsync("New Task", "Enter the Task Name");
+
+            if (result == null)
+            {
+                return;
+            }
+
+            if (taskRealm == null)
```

Comment:
Constructors can't be async, AFAIK, so I've moved it there but am using ``GetInstance()`` instead of ``GetInstanceAsync()`` now.

Reply:
That is true, I didn't think about it being async, so that's on me. 
If you want to keep the async, then you could put it in "OnAppearing" or a similar lifecycle method (obviously still checking if it was already created).
The important thing here is that it's done in one of the first methods. It's a little bit strange to see it being created in an event handler